controller/model/posture_response_handlers.go

```diff
@@ -247,3 +250,87 @@ func (handler *PostureResponseHandler) SetSdkInfo(identityId, apiSessionId strin
 		return postureData
 	})
 }
+
+type ServiceWithTimeout struct {
+	Service *Service
+	Timeout int64
+}
+
+func (handler *PostureResponseHandler) GetEndpointStateChangeAffectedServices(timeSinceLastMfa, gracePeriod time.Duration, onWake bool, onUnlock bool) []*ServiceWithTimeout {
+	affectedChecks := map[string]int64{} //check id -> timeout
+	if onWake || onUnlock {
+		queryStr := fmt.Sprintf("%s=true or %s=true", persistence.FieldPostureCheckMfaPromptOnUnlock, persistence.FieldPostureCheckMfaPromptOnWake)
+		query, err := ast.Parse(handler.env.GetStores().PostureCheck, queryStr)
+		if err != nil {
+			pfxlog.Logger().Errorf("error querying for onWake/onUnlock posture checks: %v", err)
+		} else {
+			handler.env.GetDbProvider().GetDb().View(func(tx *bbolt.Tx) error {
+				cursor := handler.env.GetStores().PostureCheck.IterateIds(tx, query)
+
+				for cursor.IsValid() {
+					if check, err := handler.env.GetStores().PostureCheck.LoadOneById(tx, string(cursor.Current())); err == nil {
+						if mfaCheck, ok := check.SubType.(*persistence.PostureCheckMfa); ok {
+							if (mfaCheck.PromptOnUnlock && onUnlock) || (mfaCheck.PromptOnWake && onWake) {
```

Comment:
nit: any reason/opportunity to make this logic a function which is then independently testable?

Reply:
You got it hoss.

Also added unit tests for it.